src/main/java/de/unijena/cheminf/mortar/model/util/ChemUtil.java

```diff
@@ -88,6 +90,46 @@ public static String createUniqueSmiles(IAtomContainer anAtomContainer) {
         return tmpSmiles;
     }
 
+    /**
+     * Returns an IAtomContainer instance which represents the molecule parsed from the SMILES string. Bond types and
+     * atom types are assigned to it (the former through kekulization) if required. Aromaticity flags are set only if
+     * there is aromaticity information present in the SMILES code, no aromaticity perception is performed here.
+     *
+     * @return IAtomContainer atom container of the molecule
+     * @param aSmilesCode SMILES representation
+     * @param shouldBeKekulized whether explicit bond orders should be assigned or "aromatic bond" can be used if present;
+     *                          does not affect aromaticity flags
+     * @param shouldAtomTypesBePerceived whether atom types should be perceived and configured
+     * @throws InvalidSmilesException if the given SMILES is invalid
+     * @throws CDKException if kekulization or atom type matching fails
+     */
+    public static IAtomContainer parseSmilesToAtomContainer(String aSmilesCode, boolean shouldBeKekulized, boolean shouldAtomTypesBePerceived)
```

Comment:
Couldn't this be used histogram code?

Reply:
Sorry, I don't really get the question. I added this method to use it in the histogram code and other places as well.

Reply:
Never mind, I misread that part.